### PR TITLE
Select Object Content API

### LINKED ISSUE #846: Support for selectObjectContent
As supported in the Java SDK:
https://docs.min.io/docs/java-client-api-reference.html#selectObjectContent

### PR DESCRIPTION
**Pull request by prakashsvmx**
Select Object Content API
`selectObjectContent(bucketName, objectName, selectOpts={}, cb)`

Fixes #846 

Tested with
- [x] Standalone server
- [x] Distributed
- [x] S3
- [x] NAS
- [x] Azure 

<summary> Test script <summary>

<details>

```
var Minio = require("../dist/main/minio")

var s3Client = new Minio.Client({
  endPoint: 's3.amazonaws.com',
  accessKey: 'YOUR-ACCESSKEYID',
  secretKey: 'YOUR-SECRETACCESSKEY'
})

const bucketName = "test-bucket"
const objectName="phonebook.csv"

s3Client.putObject(bucketName, objectName,"Name,PhoneNumber,City,Occupation\n" +
    "Sam,(949) 123-45567,Irvine,Solutions Architect\n" +
    "Vinod,(949) 123-4556,Los Angeles,Solutions Architect\n" +
    "Jeff,(949) 123-45567,Seattle,AWS Evangelist\n" +
    "Jane,(949) 123-45567,Chicago,Developer\n" +
    "Sean,(949) 123-45567,Chicago,Developer\n" +
    "Mary,(949) 123-45567,Chicago,Developer\n" +
    "Kate,(949) 123-45567,Chicago,Developer", {})
  .then(() => {
    runSelect()
  })
  .catch(e=>{
    console.log(e)
  })

function runSelect() {
  s3Client.selectObjectContent(bucketName, objectName, {
    expression: "SELECT * FROM s3object s where s.\"Name\" = 'Jane'",
    expressionType: "SQL",
    inputSerialization: {
      'CSV': {
        "FileHeaderInfo": "Use",
        RecordDelimiter: "\n",
        FieldDelimiter: ",",
      },
      'CompressionType': 'NONE'
    },
    outputSerialization: {
      'CSV': {
        RecordDelimiter: "\n",
        FieldDelimiter: ",",
      }
    },
    requestProgress: {Enabled: true}
  })
    .then((result) => {
      // verify the select query result string.
      if (result.getRecords().toString() === "Jane,(949) 123-45567,Chicago,Developer\n") { // \n for csv line ending.
        console.log("Expected Result:", result.getRecords().toString())
      } else {
        console.log("Un Expected Result:", result.getRecords().toString())
      }
    })
}

```

</details>

### CODE REVIEW

File: examples/sel-object-content.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+ * MinIO Javascript Library for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
```
Comment (kannappanr) [inline examples/sel-object-content.js:2]: Year should be 2021

**Review by ebozduman — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -213,7 +213,8 @@ The full API Reference is available here.
 * [set-object-legal-hold.js](https://github.com/minio/minio-js/blob/master/examples/set-object-legalhold.js)
 * [get-object-legal-hold.js](https://github.com/minio/minio-js/blob/master/examples/get-object-legal-hold.js)
 * [compose-object.js](https://github.com/minio/minio-js/blob/master/examples/compose-object.js)
-  
+* [select-object-content.js](https://github.com/minio/minio-js/blob/master/examples/select-object-content.js)
```
[the remark this answers is not in the document]
Comment (prakashsvmx) [reply README.md:218]: Thank you  :+1: . I have updated as per suggestions.

**Comment by prakashsvmx** [comment]
@ebozduman thank you for the review. I have updated. PTAL.

File: src/main/minio.js [hunk 3]
```diff
@@ -3357,6 +3359,97 @@ export class Client {
...
+        {"ScanRange": selOpts.scanRange}
+
+      )
+    }
+
+
+    const builder = new xml2js.Builder({rootName:'SelectObjectContentRequest', renderOpts:{'pretty':false}, headless:true})
+    const payload = builder.buildObject(config)
+
+    const md5digest = toMd5(payload)
```
[the remark this answers is not in the document]
Comment (prakashsvmx) [reply src/main/minio.js]: Yes . I have updated and tested it with AWS S3. thank you .

**Review by kanagarajkm — APPROVED** [review]

File: docs/API.md [hunk 4]
```diff
@@ -1726,6 +1724,45 @@ const composePromise = minioClient.composeObject(destOption,sourceList)
...
+
+Select contents of an object (S3 Select).
+
+__Parameters__
+
+| Param  |  Type | Description  |
+|---|---|---|
+| `bucketName`  |_string_   | Name of the bucket.  |
+| `objectName`  | _string_  | Name of the object.  |
+| `selOpts`  | _object_  |  |
```
Comment (ebozduman) [inline docs/API.md:1738]: Including this one, there are still 2 more references to `selOpts`...
line #1569 in `src/test/unit/test.js`
line #3192 in `src/test/functional/functional-tests.js`

**Review by ebozduman — CHANGES_REQUESTED** [review]

File: docs/API.md (hunk 4 above)
Comment (prakashsvmx) [reply docs/API.md]: Thank you so much.

**Review by ebozduman — DISMISSED** [review]
LGTM

**Comment by harshavardhana** [comment]
Please rebase with master @prakashsvmx

**Review by kanagarajkm — APPROVED** [review]

**Comment by ebozduman** [comment]
@prakashsvmx ,
Some issues with the build...
Please check it out

**Review by ebozduman — DISMISSED** [review]
LGTM

**Review by kanagarajkm — APPROVED** [review]

**Pull request merged**